utils/validate_pyproject.py

```diff
@@ -72,31 +72,43 @@ def update_environment_yml():
     """
     Updates environment.yml file for conda.
     """
-    from setup_requirements import install_requires
-    import yaml
 
     dir_path = os.path.dirname(os.path.realpath(__file__))
-    environment_filename = 'environment.yml'
-    file_path = os.path.join(dir_path, environment_filename)
+    sys.path.append(os.path.join(dir_path, os.pardir))
+
+    from setup_requirements import install_requires
+    import yaml
 
     # fix incompatibilities between conda and pypi
     replacements = {
         'psycopg2-binary' : 'psycopg2',
         'validate-email' : 'validate_email',
     }
-    for k in replacements:
-        install_requires = [ i.replace(k, replacements[k]) for i in install_requires]
-
+    sep = '%'  # use something forbidden in conda package names
+    pkg_string = sep.join(install_requires)
+    for (pypi_pkg_name, conda_pkg_name) in iter(replacements.items()):
+        pkg_string = pkg_string.replace(pypi_pkg_name, conda_pkg_name)
+    install_requires = pkg_string.split(sep)
```

Comment:
That's another way to do it (fine), but it seems more complicated.
Just so that I understand - are you trying to optimize for efficiency here?

Reply:
Well, from the perspective of complexity I think it's pretty much the same. But my intention was to get rid of that loop over list comprehensions to get a more linear flow which, which IMHO, would make it easier to grasp at first sight whats going on here. If you don't like it at all, I won't have any issues with reverting to the prior version.

(Efficiency: It runs about 3x faster than the loop version, but I think that should be no big deal since the loop version was fast already and I do not expect the number of dependencies to explode in the future :) )